Public/Src/Utilities/Utilities/Tracing/TextWriterEventListener.cs

```diff
@@ -134,7 +134,7 @@ public override void Dispose()
         /// <inheritdoc />
         protected override void Output(EventLevel level, int id, string eventName, EventKeywords eventKeywords, string text, bool doNotTranslatePaths = false)
         {
-            m_writer.WriteLine(level, (m_translator != null && !doNotTranslatePaths) ? m_translator.Translate(text) : text);
+            m_writer?.WriteLine(level, (m_translator != null && !doNotTranslatePaths) ? m_translator.Translate(text) : text);

```

Comment:
> m_writer? [](start = 11, length = 10)

I failed to see how `m_writer` can be null in this class. Can you explain? 

If `m_write` can never be null, then your change doesn't fix anything, no?

Reply:
From the stack, the two references in this problematic line are m_writer and m_translator.
m_translator is checked before referencing. So the only thing left is m_writer. What can make m_writer null is the m_writer.Dispose().  And this NullReferenceException happened with the console cancel event. I did try to see how it became null from code. But with all the wrappers, it is hard to see how the dispose() being called before the output(). However, with console cancel, I think there is a chance. I did see that much value to spend a lot of time to really figure out how this happen for a console cancel. If we add a check here the 

> m_writer.Dispose();
> 
> 
> since we made it optional, should we set it to null here?
> 
> Refers to: Public/Src/Utilities/Utilities/Tracing/TextWriterEventListener.cs:131 in 0fbff76. [](commit_id = 0fbff76, deletion_comment = False)

Hmm, I am not sure I see it is optional.